### PR TITLE
Don't use could_propagate in MarkDequantizationSubgraph

### PR DESCRIPTION
**Pull request by mateusztabaka**

### CODE REVIEW

File: src/common/transformations/src/transformations/low_precision/mark_dequantization_subgraph.cpp [hunk 1]
```diff
@@ -10,9 +10,24 @@
...
+    if (!node->has_evaluate() || ov::constant_folding_is_disabled(node))
+        return false;
+
+    for (size_t i = 0; i < num_inputs; i++) {
+        const auto& input = node->get_input_node_shared_ptr(i);
+        if (!is_constfoldable(input))
+            return false;
+    }
+    return true;
+}
```
Comment (eshoguli) [inline src/common/transformations/src/transformations/low_precision/mark_dequantization_subgraph.cpp:30]: Can you, please, copy exists and well tested not recursive `ngraph::could_propagate` function and just remove `ShapeOf` logic, please?

**Review by eshoguli — COMMENTED** [review]
There is comment. I got agreement with nGraph team, that we will have similar function like `ngraph::could_propagate` which will be moved to `utils` later by nGraph team later.

File: src/common/transformations/src/transformations/low_precision/mark_dequantization_subgraph.cpp (hunk 1 above)
Comment (mateusztabaka) [reply src/common/transformations/src/transformations/low_precision/mark_dequantization_subgraph.cpp]: done

**Review by eshoguli — APPROVED** [review]
LGTM

**Pull request merged**